### PR TITLE
Query node state upon parentchain block import 

### LINKED ISSUE #6: Query node state upon parentchain block import
We import a parentchain blocks here:

https://github.com/ajuna-network/worker/blob/3376deb0c15357f2339c668087ded589e7388a94/core/parentchain/block-importer/src/block_importer.rs#L122

Upon successful import, we need to query the node state: [`GameQueues`](https://github.com/ajuna-network/pallets-ajuna/blob/master/pallet-ajuna-gameregistry/src/lib.rs#L154)

An example of a storage query, including storage proof verification, can be found here: 
https://github.com/ajuna-network/worker/blob/3376deb0c15357f2339c668087ded589e7388a94/sidechain/validateer-fetch/src/validateer.rs#L60-L65

### PR DESCRIPTION
**Pull request by gaudenzkessler**
* updates the substrate sp-core
* updates teaclave
* the worker queries the state Register.GameQueues while importing parentchain blocks

closes #6

### CODE REVIEW

File: core-primitives/registry-storage/Cargo.toml [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+codec = { version = "2.0.0", default-features = false, features = ["derive"], package = "parity-scale-codec" }
+sp-std = { version = "4.0.0-dev", default-features = false, git = "https://github.com/paritytech/substrate.git", branch = "master"}
+
+#local deps
+itp-storage = { path = "../storage", default-features = false }
+
+[features]
+default = ["std"]
+std = [
+    "sp-std/std",
```
Comment (haerdib) [inline core-primitives/registry-storage/Cargo.toml:17]: Can you add `codec` here?

File: core-primitives/registry-storage/src/lib.rs [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+
+use codec::{Decode, Encode};
+use sp_std::prelude::Vec;
+
+use itp_storage::{storage_map_key, StorageHasher};
+
+pub struct RegistryStorage;
+
+// Separate the prefix from the rest because in our case we changed the storage prefix due to
+// the rebranding. With the below implementation of the `TeerexStorageKeys`, we could simply
```
Comment (haerdib) [inline core-primitives/registry-storage/src/lib.rs:11]: Copy paste error

File: core-primitives/stf-state-handler/Cargo.toml [hunk 3]
```diff
@@ -51,10 +51,10 @@ rust-base58 = { package = "rust-base58", version = "0.0.4", optional = true }
 thiserror = { version = "1.0", optional = true }
 
 # scs / integritee
-sgx-externalities = { default-features = false, git = "https://github.com/integritee-network/sgx-runtime", branch = "master", optional = true }
+sgx-externalities = { default-features = false, git = "https://github.com/ajuna-network/sgx-runtime", branch = "update-substrate", optional = true }
```
Comment (haerdib) [inline core-primitives/stf-state-handler/Cargo.toml:54]: The update-substrate branch was not merged to master on purpose?

File: core-primitives/storage-verified/src/lib.rs [hunk 4]
```diff
@@ -1,16 +1,48 @@
...
+#[cfg(all(not(feature = "std"), feature = "sgx"))]
+pub mod sgx_reexport_prelude {
+	pub use thiserror_sgx as thiserror;
+}
+
+#[cfg(all(not(feature = "std"), feature = "sgx"))]
+use crate::sgx_reexport_prelude::*;
+
+
+/// Basic storage access abstraction
```
Comment (haerdib) [inline core-primitives/storage-verified/src/lib.rs:35]: This should be a file comment, leave it as //! on top of the file.

**Review by haerdib — COMMENTED** [review]
Only minor comments, thanks a lot :)

File: core-primitives/stf-state-handler/Cargo.toml (hunk 3 above)
Comment (gaudenzkessler) [reply core-primitives/stf-state-handler/Cargo.toml:54]: i created a pr in sxg-runtimte: https://github.com/ajuna-network/sgx-runtime/pull/2

**Review by darkfriend77 — APPROVED** [review]
fine.

**Pull request merged**